plugins/woocommerce-blocks/assets/js/blocks/checkout/inner-blocks/checkout-contact-information-block/block.tsx

```diff
@@ -76,7 +132,7 @@ const Block = (): JSX.Element => {
 				values={ contactFormValues }
 				fields={ CONTACT_FORM_KEYS }
 			>
-				{ createAccountUI }
+				{ ! customerId && <CreateAccountUI /> }
```

Comment:
What would happen here if we have additional fields in Contact step? who gets rendered first? While I'm know it's gonna be a huge refactor (so don't take me for it), can't we just supply those fields as part of our fields system (so no settings in frontend) but handle rendering them manually here, just like we do with country/state fields in Address step?

Reply:
This is not a single field. It's a checkbox + password (both or one or neither). Can your API cope?

I'd guess because of the complexity of this area, and due to how conditional it is, its not a good fit for the array of fields. Am I wrong?

Reply:
> This is not a single field. It's a checkbox + password (both or one or neither). Can your API cope?

This is what we call in the biz show, a conditional field, [we don't have it implemented but we have a ticket for it](https://github.com/woocommerce/woocommerce/issues/42080). This is not a blocker and something we can do later on, but still, my couple of concerns here is how would this play with additional fields in that area, I'd suggest you just test it and see, should such fields be before this whole account thing, after them, between them...

Reply:
> What would happen here if we have additional fields in Contact step? who gets rendered first

This replaces the old create account checkbox, so its in that position.

Reply:
@senadir My main concerns over investing time into using the fields system is 
1. the current checkbox is not part of the system so we increase scope 
2. there are ongoing discussions about inclusion of username field, or moving fields to the bottom of checkout. I don't want to commit too heavy into the fields API if these components will be moved in a near future cycle.

I'll check where fields render. It will likely be below these fields.

Reply:
Yes I agree, this is not a priority, we can revisit later on. I do think at some point we need to move all custom fields we have to the system (create account, order note, terms and conditions...). This can be part of that effort.